src/layers/transform/concatenate.cu

```diff
@@ -283,16 +282,13 @@ void fp_compute_impl(
   pos += sizeof(size_t) * output_offset_list.size();
 
   // Copy tensor data to GPU
-  hydrogen::simple_buffer<unsigned char, El::Device::GPU> device_workspace(
-    l.m_workspace.size(),
-    sync_info);
-  unsigned char* device_workspace_ptr = device_workspace.data();
-  cudaMemcpyAsync(device_workspace_ptr,
-                  l.m_workspace.data(),
-                  l.m_workspace.size(),
-                  cudaMemcpyHostToDevice,
-                  stream);
-  l.m_workspace_event.record(stream);
+  gpu_lib::thrust::vector<unsigned char> device_workspace(l.m_workspace.size());
```

Comment:
`hydrogen::simple_buffer` is usually preferable to `thrust::vector`. Thrust does a lot of synchronization internally, so we should only use it when we need it (mostly for sorting).

Reply:
addressed with ba62dcb06925539311074dd079b5e52ff46524ff